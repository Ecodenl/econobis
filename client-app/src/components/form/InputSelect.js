import React from 'react';
import PropTypes from 'prop-types';

const InputSelect = props => {
    const {
        label,
        className,
        size,
        id,
        name,
        value,
        options,
        onChangeAction,
        onBlurAction,
        required,
        error,
        errorMessage,
        optionValue,
        optionName,
        readOnly,
        placeholder,
        divClassName,
        emptyOption,
    } = props;

    return (
        <div className={`form-group ${size} ${divClassName}`}>
            <label htmlFor={id} className={`col-sm-6 ${required}`}>
                {label}
            </label>
            <div className={'col-sm-6'}>
                <select
                    className={`form-control input-sm ${className}` + (error && ' has-error')}
                    id={id}
                    name={name}
                    value={value ? value : ''}
                    onChange={onChangeAction}
                    onBlur={onBlurAction}
                    readOnly={readOnly}
                >
                    {emptyOption && <option value="">{placeholder}</option>}
                    {options.map(option => {
                        return (
                            <option key={option[optionValue]} value={option[optionValue]}>
                                {option[optionName]}
                            </option>
                        );
                    })}
                </select>
            </div>
            {error && (
                <div className="col-sm-offset-6 col-sm-6">
                    <span className="has-error-message"> {errorMessage}</span>
                </div>
            )}
        </div>
    );
};

InputSelect.defaultProps = {
    divClassName: '',
    className: '',
    size: 'col-sm-6',
    readOnly: false,
    required: '',
    error: false,
    errorMessage: '',
    value: '',
    optionValue: 'id',
    optionName: 'name',
    placeholder: '',
    emptyOption: true,
};

InputSelect.propTypes = {
    label: PropTypes.string.isRequired,
    className: PropTypes.string,
    size: PropTypes.string,
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    options: PropTypes.array,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onChangeAction: PropTypes.func,
    onBlurAction: PropTypes.func,
    required: PropTypes.string,
    readOnly: PropTypes.bool,
    error: PropTypes.bool,
    errorMessage: PropTypes.string,
    emptyOption: PropTypes.bool,
    optionValue: PropTypes.string,
    optionName: PropTypes.string,
    placeholder: PropTypes.string,
};

export default InputSelect;
