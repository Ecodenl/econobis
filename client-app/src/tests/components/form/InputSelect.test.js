import React from 'react';
import { shallow } from 'enzyme';

import InputSelect from '../../../components/form/InputSelect';
const options = [{id: 1, name: 'optie 1'}, {id: 2, name: 'optie 2'}, {id: 3, name: 'optie 3'}];

test('should render InputSelect correctly', () => {
    const wrapper = shallow(<InputSelect options={options} label={"Naam"} value={1}/>);
    expect(wrapper).toMatchSnapshot();
});