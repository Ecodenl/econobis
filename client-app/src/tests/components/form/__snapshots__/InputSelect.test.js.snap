// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputSelect correctly 1`] = `
<div
  className="form-group col-sm-6"
>
  <label
    className="col-sm-6 "
  >
    Naam
  </label>
  <div
    className="col-sm-6"
  >
    <select
      className="form-control input-sm false"
      value={1}
    >
      <option>
        Keuze ...
      </option>
      <option
        value={1}
      >
        optie 1
      </option>
      <option
        value={2}
      >
        optie 2
      </option>
      <option
        value={3}
      >
        optie 3
      </option>
    </select>
  </div>
</div>
`;
